pandas/tests/io/parser/common/test_read_errors.py

```diff
@@ -224,13 +225,19 @@ def test_read_csv_wrong_num_columns(all_parsers):
         parser.read_csv(StringIO(data))
 
 
-def test_null_byte_char(all_parsers):
+def test_null_byte_char(request, all_parsers):
     # see gh-2741
     data = "\x00,foo"
     names = ["a", "b"]
     parser = all_parsers
 
-    if parser.engine == "c":
+    if parser.engine == "c" or (parser.engine == "python" and PY311):
+        if parser.engine == "python" and PY311:
+            request.node.add_marker(
+                pytest.mark.xfail(
+                    reason="In Python 3.11, this is read as an empty character not null"
```

Comment:
Curious, is there a documented change on the python side that notes this?

Reply:
I can't find anything in the whatsnew right now.

Reply:
Might be worth raising and issue on the cpython issue tracker to see if this is expected. Would save some triage work if pandas users encounter this behavior change.

Reply:
Nvm, it's actually intended. https://github.com/python/cpython/pull/28808

We can just leave it xfailed for now, and I (or someone else) can patch the Python parser to take this into account.

Reply:
Okay sounds good. Agreed to leave this xfailed and addressed later.